### PR TITLE
Add reference value helpers for *bool and *string

### PR DESCRIPTION
**Pull request by errordeveloper**
### Description

This PR adds helper for checking if a `*string` value was set and isn't an empty string, as well as helpers for checking when a `*bool` value was set and is either `true` or `false`.

This is a follow-up to #673.

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [x] All unit tests passing (i.e. `make test`)
- [ ] ~All integration tests passing (i.e. `make integration-test`)~ - blocked by #720

### CODE REVIEW

File: pkg/apis/eksctl.io/v1alpha4/types.go [hunk 1]
```diff
@@ -137,6 +137,15 @@ func NewBoolFalse() *bool {
 	return &v
 }
 
+// BoolIsSetToTrue will only return true if v is not nil and true
+func BoolIsSetToTrue(v *bool) bool { return v != nil && *v }
```
Comment (martina-if) [inline pkg/apis/eksctl.io/v1alpha4/types.go]: Maybe it could be named `FlagIsEnabled(...)`?

Comment (errordeveloper) [reply pkg/apis/eksctl.io/v1alpha4/types.go]: Yeah, maybe `IsEnabled` would be a nice and short. Word 'flag' belongs to parts where it's about CLI flags (like `pkg/ctl/cmduits`), so let's not mix up words here it's the API package :)

**Comment by errordeveloper** [comment]
@martina-if could you please take another look?

I also added another commit that fixes a minor case that we've missed in #673. I don't know if you might have fixed it in your other PR, I'm happy to drop the commit from here if you feel that we should. From the perspective of this PR that commit fixes integration tests.

**Comment by errordeveloper** [comment]
> From the perspective of this PR that commit fixes integration tests.

Actually, I said that too early! It's not true, the commit doesn't fix it. Let's work on that separately.

**Pull request merged**